native-apps/unity_ii_applink/ii_integration_dapp/src/greet_frontend/src/index.js

```diff
@@ -1,47 +1,30 @@
-import {AuthClient} from "@dfinity/auth-client"
-import {SignIdentity} from "@dfinity/agent";
-import {DelegationIdentity, Ed25519PublicKey } from "@dfinity/identity";
-
-// An incomplete Ed25519KeyIdentity with only the public key provided.
-class IncompleteEd25519KeyIdentity extends SignIdentity {
-    constructor(publicKey) {
-        super();
-        this._publicKey = publicKey;
-    }
-
-    getPublicKey () {
-        return this._publicKey;
-    }
-}
-
-function fromHexString(hexString) {
-    return new Uint8Array((hexString.match(/.{1,2}/g) ?? []).map(byte => parseInt(byte, 16))).buffer;
-}
+import {createActor, greet_backend} from "../../declarations/greet_backend";
+import {AuthClient} from "@dfinity/auth-client";
+import {HttpAgent} from "@dfinity/agent";
+import {DelegationIdentity, Ed25519PublicKey, Ed25519KeyIdentity, DelegationChain} from "@dfinity/identity";
+import {fromHexString} from "@dfinity/identity/lib/cjs/buffer";
 
-let myKeyIdentity;
-let sessionKeyIndex = -1;
+let appPublicKey;
 
 var url = window.location.href;
-sessionKeyIndex = url.indexOf("sessionkey=");
-if (sessionKeyIndex !== -1) {
-    // Parse the public session key and instantiate an IncompleteEd25519KeyIdentity.
-    var sessionkey = url.substring(sessionKeyIndex + "sessionkey=".length);
-
-    var publicKey = Ed25519PublicKey.fromDer(fromHexString(sessionkey));
-    myKeyIdentity = new IncompleteEd25519KeyIdentity(publicKey);
-} else {
-    // TODO: initialize an Ed25519KeyIdentity();
+var publicKeyIndex = url.indexOf("sessionkey=");
+if (publicKeyIndex !== -1) {
+    // Parse the public key.
+    var publicKeyString = url.substring(publicKeyString + "sessionkey=".length);
+    appPublicKey = Ed25519PublicKey.fromDer(fromHexString(publicKeyString));
 }
 
-let delegationIdentity;
+let actor = greet_backend;
+let delegationChain;
 
 const loginButton = document.getElementById("login");
 loginButton.onclick = async (e) => {
     e.preventDefault();
 
     // Create an auth client.
+    var middleKeyIdentity = Ed25519KeyIdentity.generate();
```

Comment:
IIRC, `Ed25519KeyIdentity` is not supported by subtle crypto and thus is `extractable`. I.e. the middle private key could be stolen through an XSS attack. It would be better to use the `ECDSAKeyIdentity` which is non-extractable by default.

Reply:
Sure. I'll try ECDSAKeyIdentity tomorrow. Thanks.

Reply:
Fixed at https://github.com/dfinity/examples/pull/625/commits/40620804a95c59a87dcfb268e1c50695a321769e.